src/disco/containers/InstallButton.js

```diff
@@ -29,12 +32,55 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  componentDidMount() {
+    const { dispatch, guid, installURL, slug } = this.props;
+    this.addonManager = new AddonManager(guid, installURL, this.statusChanged);
+    this.addonManager
+      .getAddon()
+      .then(() => {
+        dispatch({
+          type: 'INSTALL_STATE',
+          payload: {slug, guid, url: installURL, status: INSTALLED}});
+      }, () => {
```

Comment:
If you use catch here and update the test to use the same pattern it will work correctly.